lib/Doctrine/DBAL/Types/ConversionException.php

```diff
@@ -37,15 +41,15 @@ public static function conversionFailed($value, $toType)
      * Thrown when a Database to Doctrine Type Conversion fails and we can make a statement
      * about the expected format.
      *
-     * @param string $value
+     * @param mixed  $value
      * @param string $toType
      * @param string $expectedFormat
      *
      * @return \Doctrine\DBAL\Types\ConversionException
      */
     public static function conversionFailedFormat($value, $toType, $expectedFormat, ?Throwable $previous = null)
     {
-        $value = strlen($value) > 32 ? substr($value, 0, 20) . '...' : $value;
+        $value = is_string($value) && strlen($value) > 32 ? substr($value, 0, 20) . '...' : var_export($value, true);
```

Comment:
I'm not sure if we can `var_export()` all the time: this can easily go out of memory if somebody passes a complex object to the type conversion.